SimPEG/DCIP/DCIPUtils.py

```diff
@@ -0,0 +1,675 @@
+            #TODO consider topography and assign the closest cell center in the earth
+            if isrc == 0:
+                DATA_temp = []
+            else:
+                DATA.append(np.asarray(DATA_temp))
+                DATA_temp = []
+                indM = Utils.closestPoints(mesh2D, DATA[isrc-1][:,1:3])
+                indN = Utils.closestPoints(mesh2D, DATA[isrc-1][:,3:5])
+                rx = DCIP.RxDipole(np.c_[DATA[isrc-1][:,1:3], topoCC[indM]], np.c_[DATA[isrc-1][:,3:5], topoCC[indN]])
+                temp = np.asarray(temp)
+                if [SRC[isrc-1][0], SRC[isrc-1][1]] == [SRC[isrc-1][2], SRC[isrc-1][3]]:
+                    indA = Utils.closestPoints(mesh2D, [SRC[isrc-1][0], SRC[isrc-1][1]])
+                    tx = DCIP.SrcDipole([rx], [SRC[isrc-1][0], SRC[isrc-1][1], topoCC[indA]],[mesh.vectorCCx.max(), mesh.vectorCCy.max(), topoCC[-1]])
+                else:
+                    indA = Utils.closestPoints(mesh2D, [SRC[isrc-1][0], SRC[isrc-1][1]])
+                    indB = Utils.closestPoints(mesh2D, [SRC[isrc-1][2], SRC[isrc-1][3]])
+                    tx = DCIP.SrcDipole([rx], [SRC[isrc-1][0], SRC[isrc-1][1], topoCC[indA]],[SRC[isrc-1][2], SRC[isrc-1][3], topoCC[indB]])
+                srcLists.append(tx)
+            SRC.append(temp)
+            isrc += 1
+        elif len(temp) == 7: # SRC: X, Y and Z are provided
+            SRC.append(temp)
+            isrc += 1
+        elif len(temp) == 6: #
+            DATA_temp.append(np.r_[isrc, np.asarray(temp)])
+        elif len(temp) > 7:
+            DATA_temp.append(np.r_[isrc, np.asarray(temp)])
+
+    DATA.append(np.asarray(DATA_temp))
+    DATA_temp = []
+    indM = Utils.closestPoints(mesh2D, DATA[isrc-1][:,1:3])
+    indN = Utils.closestPoints(mesh2D, DATA[isrc-1][:,3:5])
+    rx = DCIP.RxDipole(np.c_[DATA[isrc-1][:,1:3], topoCC[indM]], np.c_[DATA[isrc-1][:,3:5], topoCC[indN]])
+    temp = np.asarray(temp)
+    if [SRC[isrc-1][0], SRC[isrc-1][1]] == [SRC[isrc-1][2], SRC[isrc-1][3]]:
+        indA = Utils.closestPoints(mesh2D, [SRC[isrc-1][0], SRC[isrc-1][1]])
+        tx = DCIP.SrcDipole([rx], [SRC[isrc-1][0], SRC[isrc-1][1], topoCC[indA]],[mesh.vectorCCx.max(), mesh.vectorCCy.max(), topoCC[-1]])
+    else:
+        indA = Utils.closestPoints(mesh2D, [SRC[isrc-1][0], SRC[isrc-1][1]])
+        indB = Utils.closestPoints(mesh2D, [SRC[isrc-1][2], SRC[isrc-1][3]])
+        tx = DCIP.SrcDipole([rx], [SRC[isrc-1][0], SRC[isrc-1][1], topoCC[indA]],[SRC[isrc-1][2], SRC[isrc-1][3], topoCC[indB]])
+    srcLists.append(tx)
+    text_file.close()
+    survey = DCIP.SurveyDC(srcLists)
+
+    # Do we need this?
+    SRC = np.asarray(SRC)
+    DATA = np.vstack(DATA)
+    survey.dobs = np.vstack(DATA)[:,-2]
+
+    # DCdata = Survey.Data(surveytest, surveytest.dobs)
+    # DCdata[src0, src0.rxList[0]]
+    return {'DCsurvey':survey, 'airind':airind, 'topoCC':topoCC, 'SRC':SRC}
+
+def readUBC_DC2DModel(fileName):
+
+    from SimPEG import np, mkvc
+    """
```

Comment:
@fourndo the doc string should go right under the function name to have it be picked up by the Python interpreter.